runtime/src/accounts_db.rs

```diff
@@ -6511,6 +6511,56 @@ impl AccountsDb {
         self.flush_slot_cache_with_clean(&[slot], None::<&mut fn(&_, &_) -> bool>, None)
     }
 
+    /// 1.13 and some 1.14 could produce legal snapshots with more than 1 append vec per slot.
+    /// This is now illegal at runtime in the validator.
+    /// However, there is a clear path to be able to support this.
+    /// So, combine all accounts from 'slot_stores' into a new storage and return it.
+    /// This runs prior to the storages being put in AccountsDb.storage
+    pub(crate) fn combine_multiple_slots_into_one_at_startup(
+        path: &Path,
+        id: AppendVecId,
+        slot: Slot,
+        slot_stores: &HashMap<AppendVecId, Arc<AccountStorageEntry>>,
+    ) -> Arc<AccountStorageEntry> {
+        let size = slot_stores
+            .iter()
+            .map(|(_id, store)| store.accounts.capacity())
+            .sum();
+        let storage = AccountStorageEntry::new(path, slot, id, size);
+
+        // get unique accounts, most recent version by write_version
+        let mut accum = HashMap::<Pubkey, StoredAccountMeta<'_>>::default();
+        slot_stores.iter().for_each(|(_id, store)| {
+            store.accounts.account_iter().for_each(|loaded_account| {
+                let loaded_write_version = loaded_account.write_version();
+                match accum.entry(*loaded_account.pubkey()) {
+                    std::collections::hash_map::Entry::Occupied(mut occupied_entry) => {
```

Comment:
*style/subjective*

Consider importing `std::collections::hash_map`.  It would still keep this unambiguous, but will make the line a bit shorter:
```suggestion
    use {
        /* ... */
        std::collections::hash_map,
        /* ... */
    };
    
                    hash_map::Entry::Occupied(mut occupied_entry) => {
```

This in only relevant if you are not applying the next suggestion.

Reply:
done